### PR TITLE
Fixes for 64-bit time

### PR DESCRIPTION
**Pull request by digitalentity**
Fix some compiling issues for 64-bit time handling.
Enable 64-bit time on targets with >128K flash

### CODE REVIEW

File: src/main/fc/fc_msp.c [hunk 1]
```diff
@@ -96,7 +96,7 @@
 #include "hardware_revision.h"
 #endif
 
-extern uint16_t cycleTime; // FIXME dependency on mw.c
+extern timeUs_t cycleTime; // FIXME dependency on mw.c
```
Comment (martinbudden) [inline src/main/fc/fc_msp.c]: This should be `timeDelta_t`

File: src/main/fc/fc_tasks.h [hunk 2]
```diff
@@ -22,7 +22,7 @@
 #include "common/time.h"
 
 void taskMainPidLoopChecker(timeUs_t currentTimeUs);
-bool taskUpdateRxCheck(timeUs_t currentTimeUs, uint32_t currentDeltaTime);
+bool taskUpdateRxCheck(timeUs_t currentTimeUs, timeUs_t currentDeltaTime);
```
Comment (martinbudden) [inline src/main/fc/fc_tasks.h]: This should be `timeDelta_t`

**Comment by martinbudden** [comment]
@digitalentity , a general comment. It seems that most of the type changes you have made should actually use `timeDelta_t`. This is a 32-bit quantity which is sufficient whenever a difference between two times is use, eg `cycletime` or in the scheduler.

**Comment by digitalentity** [comment]
@martinbudden I've made some more fixed - also fixed some of the `timeDelta_t` issues. Would be great if you can have a look.

File: src/main/fc/mw.c [hunk 3]
```diff
@@ -556,14 +556,14 @@ void filterRc(bool isRXDataNew)
...
+    const timeDelta_t currentDeltaTime = getTaskDeltaTime(TASK_SELF);
 
     if (gyroConfig()->gyroSync) {
         while (true) {
         #ifdef ASYNC_GYRO_PROCESSING
-            if (gyroSyncCheckUpdate(&gyro.dev) || ((currentDeltaTime + (micros() - currentTimeUs)) >= (getGyroUpdateRate() + GYRO_WATCHDOG_DELAY))) {
+            if (gyroSyncCheckUpdate(&gyro.dev) || (((timeUs_t)currentDeltaTime + (micros() - currentTimeUs)) >= (getGyroUpdateRate() + GYRO_WATCHDOG_DELAY))) {
         #else
-            if (gyroSyncCheckUpdate(&gyro.dev) || ((currentDeltaTime + (micros() - currentTimeUs)) >= (gyro.targetLooptime + GYRO_WATCHDOG_DELAY))) {
+            if (gyroSyncCheckUpdate(&gyro.dev) || (((timeUs_t)currentDeltaTime + (micros() - currentTimeUs)) >= (gyro.targetLooptime + GYRO_WATCHDOG_DELAY))) {
```
Comment (martinbudden) [inline src/main/fc/mw.c]: Can use `cmpTimeUs` herer.

File: src/main/io/serial_4way_stk500v2.c [hunk 4]
```diff
@@ -54,10 +54,10 @@ static uint8_t stkInBuf[16];
 #define STK_WAITCYLCES_START (STK_WAIT_TICKS / 2)  // 0.5ms
 #define STK_WAITCYLCES_EXT (STK_WAIT_TICKS * 5000) // 5s
 
-#define  WaitPinLo  while (ESC_IS_HI) { if (cmp32(micros(), timeout_timer) > 0) goto timeout; }
-#define  WaitPinHi  while (ESC_IS_LO) { if (cmp32(micros(), timeout_timer) > 0) goto timeout; }
+#define  WaitPinLo  while (ESC_IS_HI) { if (timeUs_t(micros(), timeout_timer) > 0) goto timeout; }
+#define  WaitPinHi  while (ESC_IS_LO) { if (timeUs_t(micros(), timeout_timer) > 0) goto timeout; }
```
Comment (martinbudden) [inline src/main/io/serial_4way_stk500v2.c]: Typo, this should be `cmpTimeUs`.

Comment (digitalentity) [reply src/main/io/serial_4way_stk500v2.c]: oops, thanks!

File: src/main/io/statusindicator.c [hunk 5]
```diff
@@ -69,15 +69,15 @@ void warningLedRefresh(void)
...
     warningLedTimer = now + 500000;
 }
 
 void warningLedUpdate(void)
 {
-    uint32_t now = micros();
+    timeUs_t now = micros();
 
-    if ((int32_t)(now - warningLedTimer) < 0) {
+    if ((timeDelta_t)(now - warningLedTimer) < 0) {
```
Comment (martinbudden) [inline src/main/io/statusindicator.c]: Can use `cmpTimeUs`

File: src/main/rx/ibus.c [hunk 6]
```diff
@@ -56,13 +56,13 @@ static uint8_t ibus[IBUS_BUFFSIZE] = { 0, };
...
-    uint32_t ibusTime;
-    static uint32_t ibusTimeLast;
+    timeUs_t ibusTime;
+    static timeUs_t ibusTimeLast;
     static uint8_t ibusFramePosition;
 
     ibusTime = micros();
 
-    if ((ibusTime - ibusTimeLast) > 3000)
+    if ((timeDelta_t)(ibusTime - ibusTimeLast) > 3000)
```
Comment (martinbudden) [inline src/main/rx/ibus.c]: Can use `cmpTimeUs`

**Comment by digitalentity** [comment]
This PR now compiles and works fine. It's probably safe to merge - it will enable 64-bit time on F3/F4 targets.

**Comment by martinbudden** [comment]
OK, let's get this merged in now.

**Pull request merged**